### PR TITLE
Eventhandler

### PR DESCRIPTION
**Pull request by sikhakolanu-venkatesh-ML**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+func NewEventSource(signatureHeader string, verificationFunc func(signature string, body string) bool) EventSource {
+	return EventSource{
+		signature:    signatureHeader,
+		verification: verificationFunc,
+	}
+}
+func simple_token_verification(signature string, body string) bool {
+	fmt.Println("Running verification function...")
+	if signature == "gitlabpassword" {
```
Comment (Dannana-Venkataramana-ML) [inline main.go:23]: get password from env

File: main.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		return true
+
+	}
+	return false
+
+}
+func github_verification(signature string, body string) bool {
```
Comment (Dannana-Venkataramana-ML) [inline main.go:30]: implement github and add all parser verification logic

File: main.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+func pagerduty_verification(signature string, body string) bool {
+	//Verifies that the signature received from the github event is accurate
+	return true
+
+}
+
+func main() {
+
+	http.HandleFunc("/", display)
+	http.ListenAndServe(":8090", nil)
```
Comment (Dannana-Venkataramana-ML) [inline main.go:49]: create a config file and read the port number from it that config also need to be added from kubernetes configmap

File: main.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+}
+
+func display(w http.ResponseWriter, r *http.Request) {
```
Comment (Dannana-Venkataramana-ML) [inline main.go:53]: why this needed?

File: main.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+		} else {
+			fmt.Println("Signature does not match expected signature")
+		}
+
+		PublishToPubsub(source, string(body), string(headerJson))
+
+	}
+}
+
+func getSource(headers http.Header) string {
```
Comment (Dannana-Venkataramana-ML) [inline main.go:108]: what about other sources?

**Comment by Dannana-Venkataramana-ML** [comment]
please provide proper commit message and PR message

**Comment by Dannana-Venkataramana-ML** [comment]
Where is the folder structure

File: main.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
```
Comment (Dannana-Venkataramana-ML) [inline main.go:5]: Use this github.com/sirupsen/logrus this package for logging

**Comment by Dannana-Venkataramana-ML** [comment]
Don't close PR where I left comments

File: main.go (hunk 2 above)
Comment (sikhakolanu-venkatesh-ML) [reply main.go]: Added remaining sources and all verification functions

File: main.go (hunk 4 above)
Comment (sikhakolanu-venkatesh-ML) [reply main.go]: Renamed  into eventhandler

File: main.go (hunk 5 above)
Comment (sikhakolanu-venkatesh-ML) [reply main.go]: Added remaining sources

**Review by Dannana-Venkataramana-ML — APPROVED** [review]
LGTM

File: main.go (hunk 3 above)
Comment (sikhakolanu-venkatesh-ML) [reply main.go]: Added a config file to read the port number

**Pull request merged**